src/transformers/training_args.py

```diff
@@ -1690,6 +1720,24 @@ def __post_init__(self):
             os.environ[f"{prefix}SYNC_MODULE_STATES"] = self.fsdp_config.get("sync_module_states", "true")
             os.environ[f"{prefix}USE_ORIG_PARAMS"] = self.fsdp_config.get("use_orig_params", "true")
 
+        if isinstance(self.accelerator_config, str):
+            self.accelerator_config = AcceleratorConfig.from_json_file(self.accelerator_config)
+        if self.dispatch_batches:
+            warnings.warn(
+                "Using `--dispatch_batches` is deprecated and will be removed in version 5 of 🤗 Transformers. Use"
+                " `--accelerator_config {'dispatch_batches':VALUE} instead",
+                FutureWarning,
+            )
+            self.accelerator_config["dispatch_batches"] = self.dispatch_batches
+
+        if self.split_batches:
```

Comment:
We should test for noneness here instead of relying on the default behaviour in the AccelerateConfig. We really want to set it whatever the user does. 

```suggestion
        if self.split_batches is not None:
```

Reply:
thank you